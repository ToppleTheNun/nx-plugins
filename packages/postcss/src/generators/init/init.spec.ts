import { addDependenciesToPackageJson, readJson, Tree } from "@nrwl/devkit";
import { createTreeWithEmptyWorkspace } from "@nrwl/devkit/testing";
import initGenerator from "./init";

describe("@topplethenun/nx-plugin-postcss:init", () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
  });

  describe("dependencies for package.json", () => {
    it("adds postcss if not already present", async () => {
      const existing = "existing";
      const existingVersion = "1.0.0";
      addDependenciesToPackageJson(
        tree,
        { [existing]: existingVersion },
        { [existing]: existingVersion }
      );

      await initGenerator(tree);

      const packageJson = readJson(tree, "package.json");
      expect(packageJson).toMatchSnapshot();

      expect(packageJson.devDependencies["postcss"]).toBeDefined();
      expect(packageJson.devDependencies["postcss"]).toEqual("latest");
      expect(packageJson.devDependencies["postcss-preset-env"]).toEqual(
        "latest"
      );
    });

    it("does not add postcss if already present", async () => {
      const existing = "existing";
      const existingVersion = "1.0.0";
      addDependenciesToPackageJson(
        tree,
        { [existing]: existingVersion },
        { postcss: "7", [existing]: existingVersion }
      );

      await initGenerator(tree);

      const packageJson = readJson(tree, "package.json");
      expect(packageJson).toMatchSnapshot();

      expect(packageJson.devDependencies["postcss"]).toBeDefined();
      expect(packageJson.devDependencies["postcss"]).toEqual("7");
      expect(packageJson.devDependencies["postcss-preset-env"]).toBeUndefined();
    });
  });
});
