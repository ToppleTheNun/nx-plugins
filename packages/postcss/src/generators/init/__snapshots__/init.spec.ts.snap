// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@topplethenun/nx-plugin-postcss:init dependencies for package.json adds postcss if not already present 1`] = `
Object {
  "dependencies": Object {
    "existing": "1.0.0",
  },
  "devDependencies": Object {
    "existing": "1.0.0",
    "postcss": "latest",
    "postcss-preset-env": "latest",
  },
  "name": "test-name",
}
`;

exports[`@topplethenun/nx-plugin-postcss:init dependencies for package.json does not add postcss if already present 1`] = `
Object {
  "dependencies": Object {
    "existing": "1.0.0",
  },
  "devDependencies": Object {
    "existing": "1.0.0",
    "postcss": "7",
  },
  "name": "test-name",
}
`;
